src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,55 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    public Figure getRandomFigure() {
+        //creating int with number og Figure classes to pick up one of those
+        Random randResult = new Random();
+        int figureNum = randResult.nextInt(5);
+
+        //double randomRange for multiplying with random doubles from switch-case statement
+        double randomRange = 100.0;
+
+        //getting ColorSupplier to set a random color for resulting Figure
+        ColorSupplier randomColor = new ColorSupplier();
+
+        //initializing Figure to be returned after switch-case statement
+        Figure result;
+
+        //choosing our Figure class based on random int from figureNum
+        // and then adding random parameters
+        switch (figureNum) {
+            case 0:
+                double randomSide = new Random().nextDouble() * randomRange;
+                result = new Square(randomSide);
```

Comment:
```suggestion
                return new Square(randomSide, color);
```
колір можна оголошувати над свіч кейсом і одразу робити так